lib/internal/modules/esm/module_job.js

```diff
@@ -2,26 +2,80 @@
 
 const {
   ArrayPrototypeJoin,
+  Number,
   ObjectSetPrototypeOf,
   PromiseAll,
   SafeSet,
   SafePromise,
   StringPrototypeIncludes,
   StringPrototypeMatch,
-  StringPrototypeReplace,
   StringPrototypeSplit,
 } = primordials;
 
 const { ModuleWrap } = internalBinding('module_wrap');
 
 const { decorateErrorStack } = require('internal/util');
+const { fileURLToPath } = require('url');
 const assert = require('internal/assert');
 const resolvedPromise = SafePromise.resolve();
 
 function noop() {}
 
 let hasPausedEntry = false;
 
+function extractExample(file, lineNumber) {
+  const { readFileSync } = require('fs');
+  const { parse } = require('internal/deps/acorn/acorn/dist/acorn');
+  const { findNodeAt } = require('internal/deps/acorn/acorn-walk/dist/walk');
+
+  const code = readFileSync(file, { encoding: 'utf8' });
```

Comment:
Is it OK to `readFileSync`?

Reply:
That code would run only if the process is crashing, right? I think it's perfectly fine to use `readFileSync` for that kind of things.

Reply:
This is precisely the question: I don't understand 100% yet of this error will _always_ lead to termination of the node process. If so, I also think that `readFileSync` is the way to go.

Reply:
Unless I find a way to parse the input file in a streaming fashion… Since imports typically come in the beginning of a file this would allow us to reduce the parsing work to a minimum… But maybe that's all unnecessary micro-optimizations for the crash case.

Reply:
This has been benchmarked already, see https://github.com/guybedford/cjs-module-lexer/issues/4#issuecomment-699624357. The gist of it is that streaming is not worth it for most JS files.

Reply:
Thanks for the context!

Great, so the one possible optimization would be, having read the entire file, around not parsing the entire file with with acorn, see https://github.com/nodejs/node/pull/35453#discussion_r498509474